src/main/Embeddable.tsx

```diff
@@ -2,43 +2,43 @@ import * as React from 'react'
 import {didPropsChange} from './util'
 import { ChartDesigner, CommonConfigOptions, EventManager, Region, SeatingChart, Seatsio } from '@seatsio/seatsio-types'
 
-export type EmbeddableProps<T> = {
-    onRenderStarted?: (chart: SeatingChart) => void
+export type EmbeddableProps<T > = {
+    onRenderStarted?: (chart: SeatingChart | EventManager) => void
     chartJsUrl?: string
     region: Region
 } & T
 
 export default abstract class Embeddable<T extends CommonConfigOptions> extends React.Component<EmbeddableProps<T>> {
     private container: React.RefObject<HTMLDivElement>
-    private rendering?: Promise<void>
     private chart: SeatingChart
 
+    private static seatsioBundles: { [key in Region]?: Promise<Seatsio> } = {}
+
     static defaultProps = {
         chartJsUrl: 'https://cdn-{region}.seatsio.net/chart.js'
     }
 
     constructor(props: EmbeddableProps<T>) {
         super(props);
-        this.container = React.createRef();
+        this.container = React.createRef()
     }
 
     abstract createChart (seatsio: Seatsio, config: T): SeatingChart | EventManager | ChartDesigner
 
-    async componentDidMount () {
-        if(!this.rendering) {
-            this.rendering = this.createAndRenderChart()
-        }
+    componentDidMount () {
+        !Embeddable.seatsioBundles[this.props.region] && this.createAndRenderChart()
     }
 
-    async componentDidUpdate (prevProps: EmbeddableProps<T>) {
+    componentDidUpdate (prevProps: EmbeddableProps<T>) {
         if (didPropsChange(this.props, prevProps) && this.chart) {
             this.destroyChart()
             this.createAndRenderChart()
         }
     }
 
     async createAndRenderChart () {
-        const seatsio = await this.getSeatsio()
+        window.seatsio = await this.loadSeatsio();
```

Comment:
Any reason we need to declare seatsio as a window-scoped variable? Is it to make debugging easier?

Reply:
Nope, this can be local here. But as you can see in my latest commit, window.seatsio needs to be set to undefined before loading a different region bundle.